server/apiv2/handlers/tso_keyspace_group.go

```diff
@@ -0,0 +1,137 @@
+func RegisterTSOKeyspaceGroup(r *gin.RouterGroup) {
+	router := r.Group("tso/keyspace-groups")
+	router.Use(middlewares.BootstrapChecker())
+	router.POST("", CreateKeyspaceGroups)
+	router.GET("", GetKeyspaceGroups)
+	router.GET("/:id", GetKeyspaceGroupByID)
+	router.DELETE("/:id", DeleteKeyspaceGroupByID)
+}
+
+// CreateKeyspaceGroupParams is the params for creating keyspace groups.
+type CreateKeyspaceGroupParams struct {
+	KeyspaceGroups []*endpoint.KeyspaceGroup `json:"keyspace-groups"`
+}
+
+// CreateKeyspaceGroups creates keyspace groups.
+func CreateKeyspaceGroups(c *gin.Context) {
+	svr := c.MustGet("server").(*server.Server)
+	manager := svr.GetKeyspaceGroupManager()
+	createParams := &CreateKeyspaceGroupParams{}
+	err := c.BindJSON(createParams)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, errs.ErrBindJSON.Wrap(err).GenWithStackByCause())
+		return
+	}
+
+	for _, keyspaceGroup := range createParams.KeyspaceGroups {
+		if !isValid(keyspaceGroup.ID) {
+			c.AbortWithStatusJSON(http.StatusBadRequest, "invalid keyspace group id")
+			return
+		}
+	}
+
+	err = manager.CreateKeyspaceGroups(createParams.KeyspaceGroups)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+	c.JSON(http.StatusOK, nil)
+}
+
+// GetKeyspaceGroups gets all keyspace groups.
+func GetKeyspaceGroups(c *gin.Context) {
+	svr := c.MustGet("server").(*server.Server)
+	manager := svr.GetKeyspaceGroupManager()
+	scanStart, scanLimit, err := parseLoadAllQuery(c)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	keyspaceGroups, err := manager.GetKeyspaceGroups(scanStart, scanLimit)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	c.IndentedJSON(http.StatusOK, keyspaceGroups)
+}
+
+// GetKeyspaceGroupByID gets keyspace group by id.
+func GetKeyspaceGroupByID(c *gin.Context) {
+	id, err := validateKeyspaceGroupID(c)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, "invalid keyspace group id")
+		return
+	}
+	svr := c.MustGet("server").(*server.Server)
+	manager := svr.GetKeyspaceGroupManager()
+	kg, err := manager.GetKeyspaceGroupByID(id)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	c.IndentedJSON(http.StatusOK, kg)
+}
+
+// DeleteKeyspaceGroupByID deletes keyspace group by id.
+func DeleteKeyspaceGroupByID(c *gin.Context) {
+	id, err := validateKeyspaceGroupID(c)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, "invalid keyspace group id")
+		return
+	}
+	svr := c.MustGet("server").(*server.Server)
+	manager := svr.GetKeyspaceGroupManager()
+	err = manager.DeleteKeyspaceGroupByID(id)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+	c.JSON(http.StatusOK, nil)
+}
+
+func validateKeyspaceGroupID(c *gin.Context) (uint32, error) {
```

Comment:
How about moving const maxKeyspaceGroupCount and maxKeyspaceGroupCountInUse from pkg/tso/keyspace_group_manager.go to mcs/utils/constants.go and checking if the id passed into the function is less than maxKeyspaceGroupCountInUse?

Reply:
Ok